### DISCUSSION TITLE
Incremental?

### DISCUSSION THREAD

**Discussion by microprediction** [skfolio/skfolio#4] [Ideas]
Another idea, with backtesting in mind, would be to support partial_fit for covariance estimation (and maybe warm start for portfolio optim although that's probably already in).

**Comment by HugoDelatte** [comment]
Agreed, both partial_fit and warm start add value and should be implemented

**Comment by gugar20** [comment]
I'd like to work on warm starting (using DPP) for the ConvexOptimization subclasses, focusing on MeanRisk and possibly RiskBudgeting. If available, should I file an issue with more details?

**Reply by HugoDelatte** [reply]
Yes, let’s start an issue for this.

I’m currently working on adding partial_fit support for covariance, expected returns, and prior estimators such as EWCovariance, EWMu, and the upcoming Characteristics Factor Model. I’m also implementing online counterparts for cross_val_predict, cross_val_score, and hyperparameter tuning.

For the moment, when I call partial_fit on MeanRisk, I simply trigger partial_fit on the prior estimator. If you add a warm start on top of this, it would further improve performance, so it’s a good idea.

Let’s continue this discussion in your issue.

**Reply by microprediction** [reply]
@gugar20 that would be great.

**Comment by HugoDelatte** [comment]
v0.18.0 has been released. We added:

* online learning utilities (`online_score`, `online_predict`, `OnlineGridSearch`, and `OnlineRandomizedSearch`)
* online covariance forecast evaluation utilities
* a new online regime-adjusted EW covariance estimator
* online support for EW covariance and expected return
* online support for `EmpiricalPrior`
* online support for `MeanRisk`

User Guide: https://skfolio.org/user_guide/online_learning.html
Examples: https://skfolio.org/auto_examples/online_learning/index.html

@gugar20 `MeanRisk` currently only forwards the `partial_fit` call to the expected return and covariance estimator.

In terms of speed, moment estimation was the bottleneck, so adding online support there already gives `MeanRisk` a significant boost.

Next step is to implement your suggestion: warm starts with DPP for further speed improvements.
You can go ahead with the issue/PR.

thx!
